src/helpers/algebra/cholesky.jl

```diff
@@ -5,23 +5,26 @@ using PositiveFactorizations
 
 import LinearAlgebra: BlasInt
 
-cholinv(x)           = inv(fastcholesky(x))
+cholinv(x) = inv(fastcholesky(x))
+cholinv(x::UniformScaling) = inv(x.λ) * I
 cholinv(x::Diagonal) = Diagonal(inv.(diag(x)))
-cholinv(x::Real)     = inv(x)
+cholinv(x::Real) = inv(x)
 
 function cholinv(x::AbstractMatrix{T}) where {T <: LinearAlgebra.BlasFloat}
     y = fastcholesky(x)
     LinearAlgebra.inv!(y)
     return y.factors
 end
 
-cholsqrt(x)           = Matrix(fastcholesky(x).L)
+cholsqrt(x) = Matrix(fastcholesky(x).L)
+cholsqrt(x::UniformScaling) = sqrt(x.λ) * I
 cholsqrt(x::Diagonal) = Diagonal(sqrt.(diag(x)))
-cholsqrt(x::Real)     = sqrt(x)
+cholsqrt(x::Real) = sqrt(x)
 
-chollogdet(x)           = logdet(fastcholesky(x))
+chollogdet(x) = logdet(fastcholesky(x))
+chollogdet(x::UniformScaling) = error("logdet is not defined for `UniformScaling`")
```

Comment:
`det` is defined (although only for two cases, see https://github.com/JuliaLang/julia/blob/f44be79d57c6e8c07e6abb58c39ebbbe65ad4a0e/stdlib/LinearAlgebra/src/uniformscaling.jl#L239-L247) and for `logdet` it then uses https://github.com/JuliaLang/julia/blob/f44be79d57c6e8c07e6abb58c39ebbbe65ad4a0e/stdlib/LinearAlgebra/src/generic.jl#L1701. So I propose to change this to `chollogdet(x::UniformScaling) = logdet(x)`